### PR TITLE
Fixes in Pools logic

### PR DESCRIPTION
**Pull request by osdnk**
This PR was originally changing the expanded state for LP, but I realized that before that some logic for pools needs to be fixed/added. 

This PR:

1. Fixes logic for calculating pool values - 30d profit, liquidity, annualized fee, etc. Previously we were not consequent when we are using native currencies (basically everywhere what was derived from values returned from Zerion) or just USD (from Uniswap v2 graph). Hence, while calculating we were making mistakes that resulted in incorrect values for non-USD currencies. Here I use USD for everything related to pools calculations and only "tweak" the value at the very end multiplying by the native currency rate (`useNativeCurrencyToUSD` hook).
2. Adds a logic for saving pools data in redux (I will need this for displaying values in Expanded LP state)
3. Adds a logic for calculating "fees earned". We will this for the Expanded LP state. It is adapted mostly from uniswap.info

The original changes have been moved to https://github.com/rainbow-me/rainbow/pull/1914

### CODE REVIEW

**Comment by skylarbarrera** [comment]
calculations look correct and sheet heights are basically good to go, I'll tweak a little big on Monday. 

Only issue: We need to find a way to convert the fee's earned to the selected native currency. This should also fix this on android: http://cloud.skylarbarrera.com/Screen-Shot-2021-03-26-13-48-15.19.png

File: src/redux/usersPositions.ts [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+  USER_POSITIONS,
+} from '../apollo/queries';
+import { AppDispatch, AppGetState } from '@rainbow-me/redux/store';
+import { ethereumUtils, logger } from '@rainbow-me/utils';
+
+const PRICE_DISCOVERY_START_TIMESTAMP = 1589747086;
+
+export const priceOverrides = [
+  '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48', // USDC
+  '0x6b175474e89094c44da98b954eedeac495271d0f', // DAI
```
Comment (brunobar79) [inline src/redux/usersPositions.ts:15]: Why these 2 only? What happens with all the other USD pegged stablecoins?

File: src/redux/usersPositions.ts [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+  if (position.timestamp < PRICE_DISCOVERY_START_TIMESTAMP) {
+    if (priceOverrides.includes(position?.pair?.token0.id)) {
+      position.token0PriceUSD = 1;
+    }
+    if (priceOverrides.includes(position?.pair?.token1.id)) {
+      position.token1PriceUSD = 1;
+    }
+    // WETH price
+    if (
+      position.pair?.token0.id === '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'
```
Comment (brunobar79) [inline src/redux/usersPositions.ts:63]: References

File: src/redux/usersPositions.ts [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+    ) {
+      position.token0PriceUSD = 203;
+    }
+    if (
+      position.pair?.token1.id === '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'
```
Comment (brunobar79) [inline src/redux/usersPositions.ts:68]: Same

File: src/redux/usersPositions.ts [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+      pair: pairAddress,
+      user,
+    },
+  });
+  for (const index in results.data.mints) {
+    const mint = results.data.mints[index];
+    const mintToken0 = mint.pair.token0.id;
+    const mintToken1 = mint.pair.token1.id;
+
+    // if trackign before prices were discovered (pre-launch days), hardcode stablecoins
```
Comment (brunobar79) [inline src/redux/usersPositions.ts:164]: Tracking

File: src/redux/usersPositions.ts [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+    amount0 += amount0 + parseFloat(mint.amount0);
+    amount1 += amount1 + parseFloat(mint.amount1);
+  }
+
+  for (const index in results.data.burns) {
+    const burn = results.data.burns[index];
+    const burnToken0 = burn.pair.token0.id;
+    const burnToken1 = burn.pair.token1.id;
+
+    // if trackign before prices were discovered (pre-launch days), hardcode stablecoins
```
Comment (brunobar79) [inline src/redux/usersPositions.ts:187]: Same typo

**Review by brunobar79 — APPROVED** [review]
Left some minor comments but this is 🔥🔥🔥

File: src/redux/usersPositions.ts (hunk 1 above)
Comment (osdnk) [reply src/redux/usersPositions.ts:15]: Actually, I have no idea. This part is just copied

**Comment by osdnk** [comment]
I verified the memory and this is correct

File: src/redux/usersPositions.ts (hunk 1 above)
Comment (osdnk) [reply src/redux/usersPositions.ts]: Let's leave like this I think

**Review by skylarbarrera — APPROVED** [review]

**Pull request merged**